tests/Microsoft.VisualStudio.ProjectSystem.Managed.UnitTests/Build/BuildUtilitiesTests.cs

```diff
@@ -9,6 +10,58 @@ namespace Microsoft.VisualStudio.Build
 {
     public class BuildUtilitiesTests
     {
+        [Theory]
+        [InlineData("'$(Configuration)' == ''",                                                     new[] { "Configuration" },                                new[] { "" })]
+        [InlineData("'$(BuildingInsideVisualStudio)' == 'true'",                                    new[] { "BuildingInsideVisualStudio" },                   new[] { "true" })]
+        [InlineData("'$(OS)' == 'Windows_NT'",                                                      new[] { "OS" },                                           new[] { "Windows_NT" })]
+        [InlineData("'$(Configuration)' == 'Debug'",                                                new[] { "Configuration" },                                new[] { "Debug" })]
+        [InlineData("'$(Configuration)' ==  'Debug'",                                               new[] { "Configuration" },                                new[] { "Debug" })]
+        [InlineData(" '$(Configuration)' == 'Debug' ",                                              new[] { "Configuration" },                                new[] { "Debug" })]
+        [InlineData("'$(Configuration)|$(Configuration)' == 'Debug|AnyCPU'",                        new[] { "Configuration" }, new[] { "AnyCPU" })]
+        [InlineData("'$(Configuration)|$(Platform)' == 'Debug|AnyCPU'",                             new[] { "Configuration", "Platform" },                    new[] { "Debug", "AnyCPU" })]
+        [InlineData(" '$(Configuration)|$(Platform)' == 'Debug|AnyCPU'",                            new[] { "Configuration", "Platform" },                    new[] { "Debug", "AnyCPU" })]
+        [InlineData(" '$(Configuration)|$(Platform)'  ==  'Debug|AnyCPU' ",                         new[] { "Configuration", "Platform" },                    new[] { "Debug", "AnyCPU" })]
+        [InlineData("'$(Configuration)|$(Platform)|$(TargetFramework)' == 'Debug|AnyCPU|net46'",    new[] { "Configuration", "Platform", "TargetFramework" }, new[] { "Debug", "AnyCPU", "net46" })]
+        [InlineData(" '$(Configuration)|$(Platform)|$(TargetFramework)' == 'Debug|AnyCPU|net46' ",  new[] { "Configuration", "Platform", "TargetFramework" }, new[] { "Debug", "AnyCPU", "net46" })]
+        [InlineData("'$(Configuration)|$(Platform)|$(TargetFramework)' == 'Debug|AnyCPU|'",         new[] { "Configuration", "Platform", "TargetFramework" }, new[] { "Debug", "AnyCPU", "" })]
+        public void TryCalculateConditionalProperties_ValidConditions(string condition, string[] expectedNames, string[] expectedValues)
```

Comment:
Suggest adding test cases for:

- `'$(BuildingInsideVisualStudio)'=='true'` (no spaces)
- `'$(BuildingInsideVisualStudio)' == 'True'` (different case)
- `$(BuildingInsideVisualStudio)==true` (no quotes)
- `'$(Configuration)'=='Debug'` (no spaces)